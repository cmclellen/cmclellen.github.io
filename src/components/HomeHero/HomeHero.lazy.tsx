import React, { lazy, Suspense } from 'react';

const LazyHomeHero = lazy(() => import('./HomeHero'));

const HomeHero = (props: JSX.IntrinsicAttributes & { children?: React.ReactNode; }) => (
  <Suspense fallback={null}>
    <LazyHomeHero {...props} />
  </Suspense>
);

export default HomeHero;
