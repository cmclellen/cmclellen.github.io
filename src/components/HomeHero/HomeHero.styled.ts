import styled from 'styled-components';

export const HomeHeroWrapper = styled.div`
`;
