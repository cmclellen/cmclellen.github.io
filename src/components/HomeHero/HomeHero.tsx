import React, { FC } from 'react';
import { HomeHeroWrapper } from './HomeHero.styled';

interface HomeHeroProps { }

const HomeHero: FC<HomeHeroProps> = () => (
   <HomeHeroWrapper data-testid="HomeHero">

      <div className="container py-5 shadow-4 rounded-3" >
         <h2>Welcome to my site!</h2>
         <p>
            This is a repo of a few of my experimental projects.
         </p>
      </div>

   </HomeHeroWrapper>
);

export default HomeHero;
