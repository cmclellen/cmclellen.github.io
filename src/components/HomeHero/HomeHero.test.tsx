import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import HomeHero from './HomeHero';

describe('<HomeHero />', () => {
  test('it should mount', () => {
    render(<HomeHero />);
    
    const homeHero = screen.getByTestId('HomeHero');

    expect(homeHero).toBeInTheDocument();
  });
});