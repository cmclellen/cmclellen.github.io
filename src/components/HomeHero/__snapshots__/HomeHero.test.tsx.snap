// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HomeHero /> > it should mount 1`] = `
<div
  className="sc-bdvuGq hka-DTH"
  data-testid="HomeHero"
>
  <div
    className="container py-5 shadow-4 rounded-3"
  >
    <h2>
      Welcome to my site!
    </h2>
    <p>
      This is a repo of a few of my experimental projects.
    </p>
  </div>
</div>
`;
